pkg/network/netlink/consumer.go

```diff
@@ -449,7 +449,6 @@ func (c *Consumer) GetStats() map[string]int64 {
 func (c *Consumer) Stop() {
 	if c.conn != nil {
 		c.conn.Close()
-		c.socket = nil
```

Comment:
Why was this removed?

Reply:
This was causing a crash if we closed the consumer before it started receiving events, which happens in some tests.

Reply:
Hmm. Do you have a stack trace? That feels like a problem is something is trying to use the socket after it is supposed to be closed.

Reply:
The consumer read loop in a go routine: https://github.com/DataDog/datadog-agent/blob/main/pkg/network/netlink/consumer.go#L170